### ISSUE TITLE
Improve Stack and other dialogues using tidyr - 1?

### ISSUE THREAD

**Issue by rdstern** [IDEMSInternational/R-Instat#4861]
This works, but I suggest some improvements might be useful.  It first needs a bit of discussion, and the simple data from the Climatic_Guide_datasets , with the sheet Wafric2 will do.

1. Not for this topic, but it is a mess when imported, so delete all the columns from the first where all are NA.
2. Now use **Prepare > Column: Reshape > Stack**.  At the top of the dialogue it says **Stack (melt) Columns**.  And the command seems to be **stats::reshape**.  Is this true?  I ask because the tidyr set of packages appears to have moved on.  It now calls stacking (a name I still like) **gather**.  I prefer **gather** to **melt**.  Should we change the name at the top to **Stack (gather) Columns**?  I vote we do.
3. Should we also change the actual command we use?  We say we are using the tidyr stuff.  Should we therefore update the actual command a bit, the way tidyr seems to have done? 
4. If we do update, then is this the only dialogue - for example what about unstack?  There, in the dialogue, we don't say the reshape2 word, which is **cast**.  I am happy we don't say cast, because that means nothing to me.  But tidyr says **spread** and I like that.  So I suggest we also change the current Unstack Columns to **Unstack (spread) Columns**.  
5. I also wonder about the command we use in unstack.  Should that be updated too? 
6. And in our Prepare > Column: Text we have Split and Combine.  For **Split** tidyr has **Separate**.  In the title to the dialogue itself should we have **Split (Separate) Text Columns**.  And **Combine (Unite) Text Columns**.  

I have a second set of suggestions for stack.  I am making that into a different issue, as I have been requested!

**Comment by dannyparsons** [comment]
We use `reshape2::melt` for stack and `reshape2::cast` for unstack. These are sensible functions to use because `reshape2` is written by Hadley Wickham, however not part of tidyverse. We did have long discussions about which package to use for this originally, although I don't think `tidyr` was created at that point. Hadley Wickham considers `tidyr` as a replacement for `reshape2` so it's worth considering changing. I have a memory that there is something that `reshape2::melt` can do which `tidyr::gather` can't, although I'll need to check that again.

On the names, I never felt melt and cast were useful names so would be happy to change those to gather and spread, even if we don't (yet) change the commands. I still like stack and unstack as the main names.

**Comment by dannyparsons** [comment]
So that wasn't quite correct. We use `stats::reshape` for stacking multiple column sets because it's the only stack function that does so, for a single set we use `reshape2::melt`. `tidyr::gather` cannot do multiple sets either but it is an open issue and something they want to implement.

**Comment by rdstern** [comment]
Do I get the impression that we keep the commands as they are - at least for now - but include the new names in the headings for the two dialogues?

And I also therefore add another 2 names from tidyr as mentioned above, namely Separate and Unite added to the dialogue titles for **Prepare > Column: Text > Split and Combine**.

**Comment by dannyparsons** [comment]
Yes I think that's right. Happy to add names in other places too.

**Comment by lilyclements** [comment]
Closing because we have added in these "new" `tidyr` functions - `pivot_wider` and `pivot_longer`
